src/pages/Wallet/WalletHomePage.tsx

```diff
@@ -206,6 +190,41 @@ const WalletHomePage = () => {
   )
 }
 
+const IOList = ({
+  currentAddress,
+  isOut,
+  transaction
+}: {
+  currentAddress: string
+  isOut: boolean
+  transaction: Transaction
+}) => {
+  const io = (isOut ? transaction.outputs : transaction.inputs) as Array<Output | Input>
+  const genesisTimestamp = 1231006505000
+
+  if (io && io.length > 0) {
+    if (io.every((o) => o.address === currentAddress)) {
+      return <Address key={currentAddress} hash={currentAddress} />
+    } else {
+      return (
+        <>
+          {_(io.filter((o) => o.address !== currentAddress))
+            .map((v) => v.address)
+            .uniq()
+            .value()
+            .map((v) => (
+              <Address key={v} hash={v || ''} />
+            ))}
+        </>
+      )
+    }
+  } else if (transaction.timestamp === genesisTimestamp) {
```

Comment:
The multiple `if` statements confused me a lot that I had to open the editor to realize that this code is still reachable. To improve code readability, I'd propose to replace these `if` sections with this:

```js
  if (io && io.length > 0) {
    return io.every((o) => o.address === currentAddress) ? (
      <Address key={currentAddress} hash={currentAddress} />
    ) : (
      <>
        {_(io.filter((o) => o.address !== currentAddress))
          .map((v) => v.address)
          .uniq()
          .value()
          .map((v) => (
            <Address key={v} hash={v || ''} />
          ))}
      </>
    )
  } else if (transaction.timestamp === genesisTimestamp) {
    return <TXSpecialTypeLabel>Genesis TX</TXSpecialTypeLabel>
  } else {
    return <TXSpecialTypeLabel>Mining Rewards</TXSpecialTypeLabel>
  }
```

Not a strong opinion, just an idea.

Reply:
It's very similar (question of habits I guess), I'm fine with using the ternary version ;)